daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Speedy.scala

```diff
@@ -22,7 +22,7 @@ import scala.util.control.NoStackTrace
 object Speedy {
 
   /** The speedy CEK machine. */
-  final case class Machine(
+  final class Machine(
```

Comment:
We don't create `Machine`s in a hot loop. Does this give us some benefits nevertheless?

Reply:
I really dislike using case class for stateful object. 

Basically `case class` generates for you `apply`, `unapply`, `equal`, `hashCode`,  `copy` methods and some other minor thinks.  There is a minor cost for that during allocation (probably negligible in the present case).

I do not see any of this syntactic sugar useful in the present case. Furthermore I like to see the `new` keyword when creating the object: this tells me "the object is probably stateful" (or comes from Java). 
 
It is a question of style, but I would like `case class`  to be use for algebraic data type only.

Reply:
That's a great explanation. I wasn't aware of these differences. Thanks a lot.